keps/sig-network/0006-20180314-coredns-GA-proposal.md

```diff
@@ -0,0 +1,134 @@
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    *[User Stories](#user-stories)
+    *[Implementation Details/Notes/Constraints](#implementation-details/notes/constraints) 
+
+## Summary
+
+CoreDNS is a CNCF project dedicated to the DNS service. It has developed a K8s interface that is conform to existing kube-dns specs.
+Currently, we are following the road-map defined [here](https://github.com/kubernetes/features/issues/427). CoreDNS is Beta in Kubernetes v1.10, which can be installed as an alternate to kube-dns.
+The purpose of this proposal is to graduate CoreDNS to GA and replace the current implementation of kube-dns service by CoreDNS.
+
+## Motivation
+
+* CoreDNS is a CNCF project which is in incubation status. It is dedicated to DNS Service and includes several Kubernetes related enhancement that place it as same functionality of kube-dns regarding Kubernetes usage (Federation, Autopath, Stubdomains, etc.).
+* CoreDNS has fewer moving parts than kube-dns, taking advantage of the caddy style plugin making it a single executable and single process.
+* It is written in Go, making it memory-safe (kube-dns includes dnsmasq which is not). 
+* CoreDNS has better performance than kube-dns in terms of greater QPS, lower latency, and memory. 
+* It supports a number of [use cases](https://coredns.io/2017/05/08/custom-dns-entries-for-kubernetes/) that kube-dns doesn't. As a general-purpose authoritative DNS server it has a lot of functionality that kube-dns couldn't reasonably be expected to add.
+
+
+### Goals
+
+* Have CoreDNS run as the default DNS server either after a new installation or an upgrade of the existing installation.
+* Configmap of current kube-dns would be handled by being translated into the corresponding configuration of CoreDNS.
+* [Move the CoreDNS repository to gcr.io](https://github.com/kubernetes/kubernetes/pull/59753) and define a workflow/process as to how to update the CoreDNS versions in the future.
+* Retain the existing configmap of users already using CoreDNS. 
+
+### Non-Goals
+
+* Translation of CoreDNS configmap back to kube-dns.
+* Retaining the configmap of a custom/modified CoreDNS image of CoreDNS to CoreDNS.
+* Translation of out-of configmap configuration of kube-dns.
+* Documentation on "How-to switch to CoreDNS as the default DNS service" for maintainers of other tools.
+
+## Proposal
+
+The proposed solution is to enable the selection of CoreDNS as the default DNS service over Kube-DNS during cluster deployment.
+As DNS service in Kubernetes is an add-on service that is deployed after Kubernetes itself, the deployment tools will have to comply with Kubernetes guidelines and deploy CoreDNS instead of kube-dns.
+Some of the most used deployment tools have been upgraded by CoreDNS Team, in cooperation of the owners of these tools, to be able to deploy CoreDNS:
+* kubeadm
+* kube-up
+* minikube
+* kops
```

Comment:
should we talk about "kops" here ? 
the versions are delayed - and current PR is still waiting.